math.go

```diff
@@ -14,12 +17,12 @@ func dct(width int, height int, values []uint8) (result []float32) {
 				for j := 0; j < width; j++ {
 
 					sum += float64(values[i*width+j]) *
-						math.Cos(((math.Pi*float64(u))/(2*float64(height)))*(2*float64(i)+1)) *
-						math.Cos(((math.Pi*float64(v))/(2*float64(width)))*(2*float64(j)+1))
+						math.Cos(((math.Pi*float64(u))/doubleHeight)*(2*float64(i)+1)) *
+						math.Cos(((math.Pi*float64(v))/doubleWidth)*(2*float64(j)+1))
 				}
 			}
 
-			result[u*width+v] = float32(sum)
+			result[u*width+v] = int16(sum)
```

Comment:
(Side discussion: this example isn’t doing gamma correction. Today, the `x/image/draw` doesn’t handle gamma but I think there’s some efforts to make it do so…we may end up having to handle it ourselves here before DCT.)